tests/UnitTests/ApiConnectors/BaseApiConnectorTest.php

```diff
@@ -286,4 +285,119 @@ public function testResendXmlDocumentStopsAfterMaxRetries(\Exception $exception)
         $numRetries = $this->getObjectAttribute($this->service, "numRetries");
         $this->assertEquals(0, $numRetries);
     }
+    /**
+     * @dataProvider soapFaultProvider
+     * @dataProvider errorExceptionProvider
+     */
+    public function testResendXmlDocumentStopsAfterCustomMaxRetries(\Exception $exception)
+    {
+        $this->service = $this->getMockBuilder(BaseApiConnector::class)
+            ->setConstructorArgs(
+                [
+                    $this->connection,
+                    [
+                        BaseApiConnector::CONFIG_MAX_RETRIES => 5
+                    ]
+                ]
+            )
+            ->getMockForAbstractClass();
+
+        $this->connection->expects($this->exactly($this->getMaxNumRetries() + 1))
```

Comment:
Personally, I'd change this to `exactly(6)`. The `5` is in the config above and I'd prefer explicit testing of `6` below.

Reply:
done